### PR TITLE
tests - integration/separation (`token` and `none` methods)

### PR DESCRIPTION
**Pull request by briantist**
##### SUMMARY
Originally I thought I'd be able to separate out all of the auth methods but it was a bigger job than expected, so this PR is moving `none` and `token` methods.

- Auth methods' `authenticate` method now expected to return full response, not just the token
- `token` method is special (no actual login happening) since we may not have full response; if we're looking up the token anyway (we usually are), use to it to simulate full login response
- add `.ansible-lint` file for integration tests. We're not running `ansible-lint` in CI (yet?) but my IDE's ansible plugin is using it, may affect others as well. Nice to have.
- move `token` auth to its own integration target
- remove `token` auth tests from `hashi_vault` tests
- many "token" tests were actually specific to the `hashi_vault` lookup itself; move those tests to their own file within the `hashi_vault` target. They are important tests of the plugin's functionality. Eventually, they will be among the only tests remaining there!
- update/add to the dedicated CI/test plugins
- add a policy to the test vault server that allows read access to a path that isn't populated, so that we can more properly test the code path that checks for a non-existent path (without the policy, we are given access denied instead)
- add unit tests for auth `none`
- add unit tests for auth `token` 

##### ISSUE TYPE
- Tests/CI

##### COMPONENT NAME

`tests/`

##### ADDITIONAL INFORMATION
N/A

### CODE REVIEW

File: plugins/module_utils/_hashi_vault_common.py [hunk 1]
```diff
@@ -221,6 +221,20 @@ def __init__(self, option_adapter, warning_callback):
         super(HashiVaultAuthMethodBase, self).__init__(option_adapter)
         self._warner = warning_callback
 
+    # TODO: figure out a better way to centralize imports and/or check for imports
+    # this probably doesn't really belong here, and isn't the best way to do this,
+    # but it's a start, and it's testable.
+    def check_import(self, name, has, warn=False):
```
[the remark this answers is not in the document]
Comment (briantist) [reply plugins/module_utils/_hashi_vault_common.py]: I've gone ahead and removed this method.

File: plugins/module_utils/_hashi_vault_common.py [hunk 2]
```diff
@@ -221,6 +221,20 @@ def __init__(self, option_adapter, warning_callback):
         super(HashiVaultAuthMethodBase, self).__init__(option_adapter)
         self._warner = warning_callback
 
+    # TODO: figure out a better way to centralize imports and/or check for imports
+    # this prbably doesn't really belong here, and isn't the best way to do this,
+    # but it's a start, and it's testable.
+    def check_import(self, name, has, warn=False):
+        if has:
+            return
```
[the remark this answers is not in the document]
Comment (briantist) [reply plugins/module_utils/_hashi_vault_common.py]: I've removed this method for now.

File: plugins/module_utils/_auth_method_token.py [hunk 3]
```diff
@@ -43,12 +74,26 @@ def validate(self):
...
+
+        response = None
+
         if use_token:
             client.token = token
 
-            if self._options.get_option('token_validate') and not client.is_authenticated():
-                raise HashiVaultValueError("Invalid Vault Token Specified.")
+            if lookup_self or validate:
+                self.check_import('hvac', HAS_HVAC)
```
[the remark this answers is not in the document]
Comment (briantist) [reply plugins/module_utils/_auth_method_token.py]: The method has been removed, but I'll keep that in mind for the future.

**Comment by briantist** [comment]
Thanks @webknjaz , I've updated the issue description and I believe I've addressed all your concerns. I've removed the problematic `check_import` function and I'll look at better ways to handle that n the future if needed.
